src/main/lua/assign_roles_to_user.lua

```diff
@@ -3,14 +3,26 @@
 --[[
 CREATE OR REPLACE SCRIPT ASSIGN_ROLES_TO_USER(user_name, array roles) AS
 --]]
-import(exa.meta.script_schema .. '.EXA_RLS_BASE', 'BASE')
-_roles_mask = BASE.get_roles_mask(roles)
-query([[MERGE INTO ::s.EXA_RLS_USERS U
-              USING (SELECT :u AS EXA_USER_NAME, :r AS EXA_ROLE_MASK) N
-              ON U.EXA_USER_NAME = N.EXA_USER_NAME
-              WHEN MATCHED THEN UPDATE SET U.EXA_ROLE_MASK = N.EXA_ROLE_MASK
-              WHEN NOT MATCHED THEN INSERT VALUES (N.EXA_USER_NAME, N.EXA_ROLE_MASK)]],
-              { s = exa.meta.script_schema, r = _roles_mask, u = user_name })
+import(exa.meta.script_schema .. '.EXA_RLS_BASE', 'base') -- initializes missing database objects
+import(exa.meta.script_schema .. '.EXA_IDENTIFIER', 'identifier')
+
+function assign_roles_to_user()
+    local comma_separated_role_list = "'" .. table.concat(roles, "', '") .. "'"
+    local sql = [[MERGE INTO ::s.EXA_RLS_USERS U
+                  USING (SELECT :u AS EXA_USER_NAME,
+                  (SELECT SUM(BIT_SET(0, ROLE_ID - 1))
```

Comment:
To protect against duplicate role IDs:
```suggestion
                  (SELECT SUM(DISTINCT BIT_SET(0, ROLE_ID - 1))
```

Reply:
Added.